### PR TITLE
156035399 Delete trip functionality

### PR DESCRIPTION
**Pull request by tatut**
# Added
- Possibility to delete trip from trip listing

### CODE REVIEW

File: ote/src/cljc/ote/util/collections.cljc [hunk 1]
```diff
@@ -0,0 +1,7 @@
+(ns ote.util.collections)
+
+(defn remove-by-index
+  "Remove element from vector by index."
+  [vector idx]
+  (into (subvec vector 0 idx)
+        (subvec vector (inc idx))))
```
Comment (rckopteri) [inline ote/src/cljc/ote/util/collections.cljc]: I'm kind of sure that this won't work when removing last element

File: ote/src/cljs/ote/views/route/trips.cljs [hunk 2]
```diff
@@ -138,40 +138,45 @@
...
+             [form-fields/field {:type    :time
+                                 :required? true
+                                 :update! #(update! {::transit/departure-time %})}
+              departure-time]]
+            [:div.col-md-1 {:style {:margin-left "-10px"}}
+             [exception-icon e! :departure pickup-type stop-idx row-idx]]]))))
+    stops)
+   (when can-delete?
+     [:td
+      [common/tooltip {:text "poistappa vuoro"}
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/route/trips.cljs]: poistappa suomi

File: ote/src/cljc/ote/util/collections.cljc (hunk 1 above)
Comment (tatut) [reply ote/src/cljc/ote/util/collections.cljc]: It does.

Comment (rckopteri) [reply ote/src/cljc/ote/util/collections.cljc]: I tested this. It works.

File: ote/src/cljs/ote/views/route/trips.cljs [hunk 3]
```diff
@@ -138,40 +138,45 @@
...
-                [form-fields/field {:type    :time
-                                    :required? true
-                                    :update! #(update! {::transit/departure-time %})}
-                         departure-time]]
-              [:div.col-md-1 {:style {:margin-left "-10px"}}
-                [exception-icon e! :departure pickup-type stop-idx row-idx]]]))))
-     stops)])
+    (fn [stop-idx {::transit/keys [arrival-time departure-time pickup-type drop-off-type] :as stop}]
+      (let [update! #(e! (rw/->EditStopTime row-idx stop-idx %))
+            style {:style {:padding-left     "5px"
```
Comment (rckopteri) [inline ote/src/cljs/ote/views/route/trips.cljs]: Consider mobing style to style-file

**Pull request merged**